operator/controllers/common/helperfunctions.go

```diff
@@ -41,3 +47,40 @@ func GetOldStatus(name string, statuses []klcv1alpha2.ItemStatus) apicommon.Kept
 
 	return oldstatus
 }
+
+// RecordEvent creates k8s Event and adds it to Eventqueue
+func RecordEvent(recorder record.EventRecorder, phase apicommon.KeptnPhaseType, eventType string, reconcileObject client.Object, shortReason string, longReason string, version string) {
+	msg := setEventMessage(phase, reconcileObject, longReason, version)
+	annotations := setAnnotations(reconcileObject, phase)
+	recorder.AnnotatedEventf(reconcileObject, annotations, eventType, fmt.Sprintf("%s-%s", phase.ShortName, shortReason), msg)
+}
+
+func setEventMessage(phase apicommon.KeptnPhaseType, reconcileObject client.Object, longReason string, version string) string {
+	if version == "" {
+		return fmt.Sprintf("%s: %s / Namespace: %s, Name: %s", phase.LongName, longReason, reconcileObject.GetNamespace(), reconcileObject.GetName())
+	}
+	return fmt.Sprintf("%s: %s / Namespace: %s, Name: %s, Version: %s", phase.LongName, longReason, reconcileObject.GetNamespace(), reconcileObject.GetName(), version)
```

Comment:
do we know if it is an App or Workload version?

Reply:
does not matter here, it's a parameter of the function and the function is called in a context. It can be a workload version if we call the function in workload-controller and app version if in app-controller, but it's always clear from the message, phase etc., what object (and its version) are we speaking about. In some cases, the version is not set, as the context may be EvaluationReconciliation and there we do not define version. Does this answer your question?

Reply:
The issue is that whoever reads this would need to find out if the version refers to the workload or the app based on auxiliary info. Thinking ahead of us, when there is automatic app discovery, users wouldn't know about the `KeptnApp` since they didn't create it. Hence, this could confuse the reader. Furthermore, the extra context info necessary hinders automation :/

Reply:
what do you suggest then? extra parameters for version for workload/app etc? That can grow enormously...

Reply:
This discussion can be handled in a following PR, here an issue for further discussion https://github.com/keptn/lifecycle-toolkit/issues/633

Reply:
if there are too many arguments, you might want to see if you could refactor the method. I don't have a nice solution on top of my head right now but maybe instantiating this with a receiver and use the optional functional pattern could yield the signature `RecordEvent(o ...EventOptions)`. I don't know if this is feasible though.

Reply:
This is also a good candidate for refactoring PR

Reply:
we should exactly define what will be part of the events, because now it's only guessing